format/avro/codecs/long.go

```diff
@@ -0,0 +1,34 @@
+package codecs
+
+import (
+	"github.com/wader/fq/format/avro/schema"
+	"github.com/wader/fq/pkg/decode"
+)
+
+const intMask = byte(127)
+const intFlag = byte(128)
+
+// readLong reads a variable length zig zag long from the current position in decoder
+func VarZigZag(d *decode.D) int64 {
+	var value uint64
+	var shift uint
+	for d.NotEnd() {
+		b := byte(d.U8())
+		value |= uint64(b&intMask) << shift
+		if b&intFlag == 0 {
+			return int64(value>>1) ^ -int64(value&1)
+		}
+		shift += 7
+	}
+	panic("unexpected end of data")
```

Comment:
In decoder i've used two different kinds of a panics, one kind is expected things like io error, malformed files etc and anything else (like a string) are treated as bugs and will crash fq. For expected things there are helper functions like `d.IOPanic`, `d.Errorf` and `d.Fatalf`.

Reply:
Woops, fixed with d.Fatalf